spec/src/main/resources/META-INF/smithy/bsp/extensions/rust.smithy

```diff
@@ -0,0 +1,393 @@
+    env: RustEnvironmentVariables
+    /// An absolute path which is used as a value of `OUT_DIR` environmental
+    /// variable when compiling current package.
+    outDirUrl: URI
+    /// File path to compiled output of a procedural macro crate.
+    /// Procedural macros are macros that generate code at compile time.
+    /// Contains files with file extensions: `.dll`, `.so` or `.dylib`.
+    procMacroArtifact: String
+}
+
+list RustPackageEnabledFeatures {
+    member: String
+}
+
+list RustTargets {
+    member: RustBuildTarget
+}
+
+/// This structure is embedded in the `data?: BuildTargetData` field, when the
+/// `dataKind` field contains "rust".
+@dataKind(kind: "rust", extends: [BuildTargetData])
+structure RustBuildTarget {
+    /// The name of the target.
+    @required
+    name: String
+    /// Path to the root module of the crate.
+    @required
+    crateRootUrl: URI
+    /// A target's kind.
+    @required
+    kind: RustTargetKind
+    /// Type of output that is produced by a crate during the build process.
+    /// The crate type determines how the source code is compiled.
+    crateTypes: RustCrateTypes
+    /// The Rust edition of the target.
+    @required
+    edition: RustEdition
+    /// Whether or not this target has doc tests enabled, and
+    /// the target is compatible with doc testing.
+    @required
+    doctest: Boolean
+    /// A sequence of required features.
+    requiredFeatures: RustPackageRequiredFeatures
+}
+
+list RustCrateTypes {
+    member: RustCrateType
+}
+
+/// Crate types (`lib`, `rlib`, `dylib`, `cdylib`, `staticlib`) are listed for
+/// `lib` and `example` target kinds. For other target kinds `bin` crate type is listed.
+@enumKind("closed")
+intEnum RustCrateType {
+    BIN = 1
+    LIB = 2
+    RLIB = 3
+    DYLIB = 4
+    CDYLIB = 5
+    STATICLIB = 6
+    PROC_MACRO = 7
+    UNKNOWN = 8
+}
+
+@enumKind("closed")
+intEnum RustTargetKind {
+    /// For lib targets.
+    LIB = 1
+    /// For binaries.
+    BIN = 2
+    /// For integration tests.
+    TEST = 3
+    /// For examples.
+    EXAMPLE = 4
+    /// For benchmarks.
+    BENCH = 5
+    /// For build scripts.
+    CUSTOMBUILD = 6
+    /// For unknown targets.
+    UNKNOWN = 7
+}
+
+list RustPackageRequiredFeatures {
+    member: String
+}
+
+list RustFeatures {
+    member: RustFeature
+}
+
+structure RustFeature {
+    /// Name of the feature.
+    @required
+    name: String
+    /// Feature's dependencies.
+    @required
+    dependencies: RustFeatureDependencies
+}
+
+list RustFeatureDependencies {
+    member: String
+}
+
+structure RustCfgOptions {
+    /// `cfgs` in Rust can take one of two forms: "cfg1" or "cfg2=\"string\"".
+    /// The `cfg` is split by '=' delimiter and the first half becomes key and
+    /// the second is aggregated to the value in `keyValueOptions`.
+    keyValueOptions: KeyValueOptions
+    /// A sequence of first halves after splitting `cfgs` by '='.
+    nameOptions: RustCfgOptionsNames
```

Comment:
wouldn't it be simpler to translate `"cfg1"` to a map `"cfg" -> []`? this way we could have a single representation

Reply:
I think we can do that